src/py21cmmc/core.py

```diff
@@ -1151,3 +1151,138 @@ def get_cl(self, cosmo, l_max=-1):
+            "tau",
+            "tau_err"
+        ),
+        cache_dir=None,
+        version="latest",
+        store=[],
+        *args,
+        **kwargs,
+    ):
+
+        super().__init__(*args, **kwargs)
+        self.name = str(name)
+        self.ctx_variables = ctx_variables
+        
+        try:
+            from py21cmemu import Emulator, properties
+        except:
+            print("Could not load py21cmemu. Make sure it is installed properly.")
+        self.astro_param_keys = ("F_STAR10",
+                                "ALPHA_STAR",
+                                "F_ESC10",
+                                "ALPHA_ESC",
+                                "M_TURN",
+                                "t_STAR",
+                                "L_X",
+                                "NU_X_THRESH",
+                                "X_RAY_SPEC_INDEX")
+        if astro_params is not None:
+            if isinstance(astro_params, p21.AstroParams):
+                self.astro_params = astro_params
+            else:
+                self.astro_params = p21.AstroParams(astro_params)
+        else:
+            self.astro_params = p21.AstroParams()
+
+        self.cosmo_params = p21.CosmoParams(properties.COSMO_PARAMS)
+        self.flag_options = p21.FlagOptions(properties.FLAG_OPTIONS)
+        self.user_params = p21.UserParams(properties.USER_PARAMS)
+        self.global_params = global_params or {}
+        self.io_options  = {
+                "store": store,  # which summaries to store
+                "cache_dir": cache_dir,  # where the stored data will be written
+        }
+
+        self.emulator = Emulator(version=version)
+
+    def _update_params(self, params):
+        """
+        Update all the parameter structures which get passed to the driver.
+
+        Parameters
+        ----------
+        params :
+            Parameter object from cosmoHammer
+        """
+        ap_dict = copy.copy(self.astro_params.self)
+
+        ap_dict.update(
+            **{
+                k: getattr(params, k)
+                for k, v in params.items()
+                if k in self.astro_params.defining_dict
+            }
+        )
+
+        return p21.AstroParams(**ap_dict)
+
+    def build_model_data(self, ctx):
+        """Compute all data defined by this core and add it to the context."""
+        # Update parameters
+        logger.debug(f"Updating parameters: {ctx.getParams()}")
+        astro_params = self._update_params(ctx.getParams())
+        logger.debug(f"AstroParams: {astro_params}")
+        # Take only needed AstroParams
+        input_dict = {k: getattr(astro_params, k)
+                for k in self.astro_param_keys}
+
+        # Call 21cmEMU wrapper which returns a dict
+        theta, outputs, errors = self.emulator.predict(astro_params=input_dict)
+        if self.io_options['cache_dir'] is not None:
+            outputs.write(fname = self.io_options['cache_dir'],#Does this include the fname as well?
```

Comment:
Typically the `cache_dir` would not include the filename, only the directory. I think the idea is that the fname might be specified dynamically.